### PR TITLE
Add source folder variable to fastapi fix 12118

### PR DESCRIPTION
**Pull request by rk0n**
ref: #12118 

### What was done:
- Add Codegen SourceFolder Variable to FastAPI Codegen

### How to test changes:
- Edit soureFolder Variable in bin/configs/python-fastapi.yaml
- generate examples
- check different folder in example

### PR checklist
 
- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [X] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [X] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.
@taxpon @frol @mbohlool @cbornet @kenjones-cisco @tomplus @spacether

### CODE REVIEW

**Comment by spacether** [comment]
Hey there, your sample generation test is failing. Can you please rebase on master and regenerate your samples?

File: samples/server/petstore/python-fastapi/setup.cfg [hunk 1]
```diff
@@ -11,13 +11,10 @@ classifiers =
...
-install_requires =
-    fastapi[all]
-setup_requires =
-    setuptools
-package_dir=
-    =src
-packages=find_namespace:
+install_requires = fastapi[all]
+setup_requires = setuptools
+package_dir = =src
```
Comment (spacether) [inline samples/server/petstore/python-fastapi/setup.cfg:16]: Should this have a double equals here?

File: modules/openapi-generator/src/main/resources/python-fastapi/setup_cfg.mustache [hunk 2]
```diff
@@ -11,13 +11,10 @@ classifiers =
...
-install_requires =
-    fastapi[all]
-setup_requires =
-    setuptools
-package_dir=
-    =src
-packages=find_namespace:
+install_requires = fastapi[all]
+setup_requires = setuptools
+package_dir = ={{sourceFolder}}
```
Comment (spacether) [inline modules/openapi-generator/src/main/resources/python-fastapi/setup_cfg.mustache:16]: Should this have a double equals here?

Comment (rk0n) [reply modules/openapi-generator/src/main/resources/python-fastapi/setup_cfg.mustache:16]: According to https://setuptools.pypa.io/en/latest/userguide/declarative_config.html?highlight=package_dir#using-a-src-layout this option needs double equals.

**Comment by rk0n** [comment]
> Hey there, your sample generation test is failing. Can you please rebase on master and regenerate your samples?

@spacether Will do it when I'm home again at the end of this week!

**Review by spacether — APPROVED** [review]
This looks good. Thank you for your PR!

**Pull request merged**